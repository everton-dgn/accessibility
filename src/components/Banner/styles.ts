import styled, { css } from 'styled-components'
import Layouts from 'layouts'
import { Waves } from 'components/Svg/Waves'

export const Header = styled.header.attrs({ role: 'heading' })`
  width: 100%;
  display: flex;
  flex-direction: column;
`

export const Container = styled(Layouts.Row)`
  ${({ theme }) => css`
    background-color: ${theme.colors.grayLight};
  `}
  min-height: 31.3rem;
  position: relative;
`

export const ImgPinkBalloon = styled.svg.attrs({ alt: 'Balão Rosa' })`
  display: none;
  position: absolute;
  left: -2.5rem;
  top: -2rem;

  ${({ theme }) => css`
    opacity: 0;
    animation: ${theme.animations.fadeInMove('0', '-30px', '0')} 0.8s ease
      forwards;

    ${theme.breakPoint(800)} {
      display: block;
    }
  `}
`

export const ImgPurpleBalloon = styled.svg.attrs({ alt: 'Balão Roxo' })`
  display: none;
  position: absolute;
  right: -4.5rem;
  bottom: -3rem;

  ${({ theme }) => css`
    opacity: 0;
    animation: ${theme.animations.fadeInMove('0', '30px', '0')} 0.8s ease
      forwards;

    ${theme.breakPoint(800)} {
      display: block;
    }
  `}
`

export const Row = styled(Layouts.Row)`
  flex-direction: column-reverse;

  ${({ theme }) => theme.breakPoint(800)} {
    flex-direction: row;
  }
`

export const WrapperInfo = styled(Layouts.Col)`
  width: auto;
  justify-content: center;
  padding: 2rem;

  ${({ theme }) => theme.breakPoint(600)} {
    padding: 3rem 5rem;
  }

  ${({ theme }) => theme.breakPoint(998)} {
    padding: 6rem 10rem;
  }
`

export const Title = styled(Layouts.Col)`
  ${({ theme }) => css`
    opacity: 0;
    animation: ${theme.animations.fadeInMove('0', '-40px', '0')} 0.8s ease
      forwards 0.2s;
    font-size: ${theme.font.sizes.xLarge};
    font-weight: ${theme.font.bold};
  `}
`

export const Description = styled(Layouts.Col)`
  ${({ theme }) => css`
    opacity: 0;
    animation: ${theme.animations.fadeInMove('0', '30px', '0')} 0.8s ease
      forwards 0.2s;
    font-size: ${theme.font.sizes.medium};
    font-weight: ${theme.font.normal};
    color: ${theme.colors.gray};
  `}
  line-height: 1.5;
  padding: 1.2rem 0;
`

export const Question = styled(Layouts.Col)`
  ${({ theme }) => css`
    opacity: 0;
    animation: ${theme.animations.fadeInMove('30px', '0', '0')} 0.8s ease
      forwards 0.4s;
    font-size: ${theme.font.sizes.medium};
    font-weight: ${theme.font.bold};
  `}
  padding: 1rem 0 1.5rem 0;
`

export const BtnGroup = styled(Layouts.Row)`
  gap: 1.6rem;
  flex-wrap: wrap;

  ${({ theme }) => css`
    .btn0 {
      opacity: 0;
      animation: ${theme.animations.fadeInMove('0', '30px', '0')} 0.8s ease
        forwards 0.7s;
    }

    .btn1 {
      opacity: 0;
      animation: ${theme.animations.fadeInMove('0', '30px', '0')} 0.8s ease
        forwards 0.9s;
    }
  `}
`

export const WrapperImg = styled(Layouts.Row)`
  position: relative;
  width: 100%;
  min-width: initial;
  max-width: initial;
  aspect-ratio: 1/0.8;

  ${({ theme }) => css`
    opacity: 0;
    animation: ${theme.animations.fadeIn} 2s ease forwards;

    ${theme.breakPoint(800)} {
      min-width: 43.3rem;
      max-width: 43.3rem;
    }
  `}

  & > div {
    position: initial !important;
  }

  img {
    object-fit: cover;
  }
`

export const WavesSvg = styled(Waves)`
  display: none;
  position: absolute;
  z-index: ${({ theme }) => theme.layers.base};
  left: -0.2rem;
  height: 100%;
  width: auto;

  ${({ theme }) => theme.breakPoint(800)} {
    display: flex;
  }
`

export const Footer = styled(Layouts.Row)`
  margin-top: 2rem;
  padding: 0 2rem 3rem 2rem;

  ${({ theme }) => css`
    opacity: 0;
    animation: ${theme.animations.fadeInMove('30px', '0', '0')} 0.8s ease
      forwards 1.1s;

    ${theme.breakPoint(800)} {
      padding: 0 0 3rem 0;
    }
  `}
`
