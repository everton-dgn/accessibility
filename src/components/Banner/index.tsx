import { ReactNode } from 'react'
import Image from 'next/image'
import { useRouter } from 'next/router'
import * as C from 'components'
import * as S from 'components/Banner/styles'

export type BannerProps = {
  title: string
  description: string
  question?: string
  img: string
  btn?: { textBtn: string; route: string }[]
  children?: ReactNode
}

const Banner = ({
  title,
  description,
  question,
  img,
  btn,
  children
}: BannerProps) => {
  const router = useRouter()

  return (
    <S.Header role="heading">
      <S.Container>
        <S.ImgPinkBalloon
          aria-hidden="true"
          width="62"
          height="62"
          viewBox="0 0 62 62"
          fill="none"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 31C0 13.8792 13.8792 0 31 0C48.1208 0 62 13.8792 62 31C62 48.1208 48.1208 62 31 62H0L0 31Z"
            fill="#FF5B58"
          />
        </S.ImgPinkBalloon>

        <S.Row>
          <S.WrapperInfo>
            <S.Title>{title}</S.Title>
            <S.Description>{description}</S.Description>
            {!!question && <S.Question>{question}</S.Question>}

            {!!btn && (
              <S.BtnGroup>
                {btn?.map(({ textBtn, route }, i) => (
                  <C.Button
                    key={textBtn}
                    color="blue"
                    size="large"
                    text={textBtn}
                    onClick={() => router.push(route)}
                    aria-label={textBtn}
                    className={`btn${i}`}
                  />
                ))}
              </S.BtnGroup>
            )}
          </S.WrapperInfo>

          <S.WrapperImg>
            <Image
              src={img}
              alt="Banner Principal"
              width="433"
              height="313"
              priority={true}
            />
            <S.WavesSvg />
          </S.WrapperImg>
        </S.Row>

        <S.ImgPurpleBalloon
          aria-hidden="true"
          width="100"
          height="100"
          viewBox="0 0 100 100"
          fill="none"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 50C0 22.3858 22.3858 0 50 0C77.6142 0 100 22.3858 100 50C100 77.6142 77.6142 100 50 100H0L0 50Z"
            fill="#334BC8"
          />
        </S.ImgPurpleBalloon>
      </S.Container>

      {!!children && <S.Footer>{children}</S.Footer>}
    </S.Header>
  )
}

export default Banner
