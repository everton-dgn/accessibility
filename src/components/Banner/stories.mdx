import { Canvas, Meta, Story } from '@storybook/addon-docs'
import Banner from '.'
import * as stories from './stories'

<Meta
  title="Design System/Banner"
  component={Banner}
  parameters={{
    controls: { sort: 'requiredFirst' },
    layout: 'fullscreen'
  }}
/>

# Banner

General purpose banner.

## Default

The banner component renders title, description, question, image, buttons, and a children.

<Canvas>
  <Story story={stories.Default} />
</Canvas>
