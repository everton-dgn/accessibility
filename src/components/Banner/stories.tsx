import Link from 'next/link'
import { Story } from '@storybook/react'
import * as S from 'pages/pagina_exemplo_1/styles'
import Banner, { BannerProps } from '.'

const Template: Story<BannerProps> = args => (
  <S.WrapperBanner>
    <S.Container>
      <Banner {...args}>
        <S.Paragraph>
          Qualquer dúvida que você tiver é só nos chamar no nosso número de
          <Link href="/boas-vindas" passHref>
            <S.LinkZap>WhatsApp.</S.LinkZap>
          </Link>
        </S.Paragraph>
      </Banner>
    </S.Container>
  </S.WrapperBanner>
)

export const Default = Template.bind({})

Default.args = {
  title: 'Bem-vindo Marcio,',
  description:
    'Para você aproveitar nossas oportunidades primeiro você precisa estar dentro de um grupo, estamos aqui para lhe ajudar.',
  question: 'O que você prefere?',
  img: '/imgs/persons.jpg',
  btn: [
    { textBtn: 'Criar um grupo', route: '/boas-vindas' },
    { textBtn: 'Entrar em um grupo', route: '/boas-vindas' }
  ]
}
