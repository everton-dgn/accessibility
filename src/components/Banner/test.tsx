import { screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { renderWithTheme } from 'utils/tests/helpers'
import Banner from '.'
const useRouter = jest.spyOn(require('next/router'), 'useRouter')
const push = jest.fn()

useRouter.mockImplementation(() => ({
  push,
  query: '',
  asPath: '',
  route: '/'
}))

describe('<Banner />', () => {
  beforeEach(() => {
    renderWithTheme(
      <Banner
        title="Bem-vindo Marcio,"
        description="Descrição de exemplo."
        question="O que você prefere?"
        img="/imgs/persons.jpg"
        btn={[
          { textBtn: 'Criar um grupo', route: '/criar-grupo' },
          { textBtn: 'Entrar em um grupo', route: '/boas-vindas' }
        ]}
      />
    )
  })

  it('should render the banner', () => {
    const banner = screen.getByRole('banner')

    expect(banner).toBeInTheDocument()
    expect(banner).toMatchSnapshot()
  })

  it('should render 3 texts and 1 image', () => {
    const textTitle = screen.getByText('Bem-vindo Marcio,')
    const textDescription = screen.getByText('Descrição de exemplo.')
    const textQuestion = screen.getByText('O que você prefere?')
    const img = screen.getByAltText('Banner Principal')

    expect(textTitle).toBeInTheDocument()
    expect(textDescription).toBeInTheDocument()
    expect(textQuestion).toBeInTheDocument()
    expect(img).toBeInTheDocument()
  })

  it('should render 2 botões', () => {
    const btn1 = screen.getByRole('button', { name: 'Criar um grupo' })
    const btn2 = screen.getByRole('button', { name: 'Entrar em um grupo' })

    expect(btn1).toBeInTheDocument()
    expect(btn2).toBeInTheDocument()
  })

  it('should call a route by clicking the button', async () => {
    userEvent.click(
      await screen.getByRole('button', { name: 'Criar um grupo' })
    )

    expect(push).toHaveBeenCalledWith('/criar-grupo')

    userEvent.click(
      await screen.getByRole('button', { name: 'Entrar em um grupo' })
    )

    expect(push).toHaveBeenCalledWith('/boas-vindas')
  })
})
