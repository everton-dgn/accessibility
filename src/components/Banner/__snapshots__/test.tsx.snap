// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> should render the banner 1`] = `
.c11 {
  cursor: pointer;
  border-radius: 0.4rem;
  font-weight: 400;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  border: 2px solid transparent;
  text-shadow: 0 0 0.3rem rgb(0 0 0 / 60%);
  -webkit-transition: background-color 0.15s ease-in-out;
  transition: background-color 0.15s ease-in-out;
  width: 100%;
  background-color: #334BC8;
  height: -webkit-min-content;
  height: -moz-min-content;
  height: min-content;
  font-size: 1.4rem;
  padding: 1.6rem 3.2rem;
}

.c11:hover {
  background-color: #5872ff;
}

.c11:disabled,
.c11:disabled:hover {
  opacity: 0.5;
  cursor: not-allowed;
  background-color: #a1a4b0;
}

.c11:focus {
  border: 2px solid #fff;
  box-shadow: 0 0 0 2px #222;
}

.c11 span {
  color: #fff;
}

.c11 svg + span {
  margin-left: 1rem;
}

.c11 svg {
  width: 2rem;
  min-width: 2rem;
  height: 2rem;
  min-height: 2rem;
  color: #fff;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
}

.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  background-color: #FAFAFA;
  min-height: 31.3rem;
  position: relative;
}

.c3 {
  display: none;
  position: absolute;
  left: -2.5rem;
  top: -2rem;
  -webkit-filter: drop-shadow(3px 7px 18px #0a0d277a);
  filter: drop-shadow(3px 7px 18px #0a0d277a);
  opacity: 0;
  -webkit-animation: jfPkyz 0.8s ease forwards;
  animation: jfPkyz 0.8s ease forwards;
}

.c14 {
  display: none;
  position: absolute;
  right: -4.5rem;
  bottom: -3rem;
  -webkit-filter: drop-shadow(-3px 7px 18px #0a0d277a);
  filter: drop-shadow(-3px 7px 18px #0a0d277a);
  opacity: 0;
  -webkit-animation: hngCSK 0.8s ease forwards;
  animation: hngCSK 0.8s ease forwards;
}

.c4 {
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
}

.c6 {
  width: auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 2rem;
}

.c7 {
  opacity: 0;
  -webkit-animation: bNBXqU 0.8s ease forwards 0.2s;
  animation: bNBXqU 0.8s ease forwards 0.2s;
  font-size: 2.8rem;
  font-weight: 500;
}

.c8 {
  opacity: 0;
  -webkit-animation: hngCSK 0.8s ease forwards 0.2s;
  animation: hngCSK 0.8s ease forwards 0.2s;
  font-size: 1.6rem;
  font-weight: 400;
  color: #7A7A7A;
  line-height: 1.5;
  padding: 1.2rem 0;
}

.c9 {
  opacity: 0;
  -webkit-animation: bpRhcC 0.8s ease forwards 0.4s;
  animation: bpRhcC 0.8s ease forwards 0.4s;
  font-size: 1.6rem;
  font-weight: 500;
  padding: 1rem 0 1.5rem 0;
}

.c10 {
  gap: 1.6rem;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c10 .btn0 {
  opacity: 0;
  -webkit-animation: hngCSK 0.8s ease forwards 0.7s;
  animation: hngCSK 0.8s ease forwards 0.7s;
}

.c10 .btn1 {
  opacity: 0;
  -webkit-animation: hngCSK 0.8s ease forwards 0.9s;
  animation: hngCSK 0.8s ease forwards 0.9s;
}

.c12 {
  position: relative;
  width: 100%;
  min-width: initial;
  max-width: initial;
  aspect-ratio: 1/0.8;
  opacity: 0;
  -webkit-animation: dCyGXD 2s ease forwards;
  animation: dCyGXD 2s ease forwards;
}

.c12 > span {
  position: initial !important;
}

.c12 img {
  object-fit: cover;
}

.c13 {
  display: none;
  position: absolute;
  z-index: 10;
  left: -0.2rem;
  height: 100%;
  width: auto;
}

@media screen and (min-width:480px) {
  .c11 {
    width: -webkit-fit-content;
    width: -moz-fit-content;
    width: fit-content;
  }
}

@media screen and (min-width:800px) {
  .c3 {
    display: block;
  }
}

@media screen and (min-width:800px) {
  .c14 {
    display: block;
  }
}

@media screen and (min-width:800px) {
  .c4 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:600px) {
  .c6 {
    padding: 3rem 5rem;
  }
}

@media screen and (min-width:998px) {
  .c6 {
    padding: 6rem 10rem;
  }
}

@media screen and (min-width:800px) {
  .c12 {
    min-width: 43.3rem;
    max-width: 43.3rem;
  }
}

@media screen and (min-width:800px) {
  .c13 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<header
  class="c0"
  role="banner"
>
  <div
    class="c1 c2"
  >
    <svg
      aria-hidden="true"
      class="c3"
      fill="none"
      height="62"
      viewBox="0 0 62 62"
      width="62"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 31C0 13.8792 13.8792 0 31 0C48.1208 0 62 13.8792 62 31C62 48.1208 48.1208 62 31 62H0L0 31Z"
        fill="#FF5B58"
      />
    </svg>
    <div
      class="c1 c4"
    >
      <div
        class="c5 c6"
      >
        <h1
          class="c5 c7"
        >
          Bem-vindo Marcio,
        </h1>
        <h2
          class="c5 c8"
        >
          Descrição de exemplo.
        </h2>
        <div
          class="c5 c9"
        >
          O que você prefere?
        </div>
        <div
          class="c1 c10"
        >
          <button
            aria-label="Criar um grupo"
            class="c11 btn0"
            color="blue"
          >
            <span>
              Criar um grupo
            </span>
          </button>
          <button
            aria-label="Entrar em um grupo"
            class="c11 btn1"
            color="blue"
          >
            <span>
              Entrar em um grupo
            </span>
          </button>
        </div>
      </div>
      <div
        class="c1 c12"
      >
        <img
          alt="Banner Principal"
          height="313"
          src="/imgs/persons.jpg"
          style="height: 100%;"
          width="433"
        />
        <svg
          class="c13"
          height="313px"
          viewBox="0 0 3.62 72.91"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1.29 5.73c-0.31,-0.7 -0.47,-1.35 -0.48,-2.05 0,-0.69 0.14,-1.44 0.41,-2.06 0.27,-0.62 0.65,-1.12 1.04,-1.62l-2.26 0 0 72.91 2.2 0c0.57,-0.93 1.13,-1.86 1.32,-2.86 0.18,-1 -0.01,-2.07 -0.52,-3 -0.51,-0.93 -1.34,-1.71 -1.84,-2.81 -0.51,-1.1 -0.71,-2.51 -0.16,-3.91 0.54,-1.39 1.81,-2.77 2.32,-4.01 0.52,-1.24 0.28,-2.34 -0.09,-3.23 -0.38,-0.89 -0.89,-1.58 -1.42,-2.52 -0.53,-0.94 -1.07,-2.14 -1.09,-3.19 -0.01,-1.06 0.5,-1.96 1.06,-2.9 0.56,-0.94 1.17,-1.9 1.49,-2.86 0.31,-0.96 0.32,-1.91 0.06,-2.72 -0.26,-0.81 -0.78,-1.49 -1.28,-2.27 -0.49,-0.79 -0.96,-1.68 -1.17,-2.59 -0.22,-0.91 -0.19,-1.83 0.11,-2.66 0.29,-0.84 0.86,-1.58 1.38,-2.42 0.52,-0.83 1.01,-1.77 1.15,-2.74 0.14,-0.98 -0.06,-2 -0.48,-2.92 -0.42,-0.92 -1.06,-1.74 -1.46,-2.36 -0.4,-0.63 -0.56,-1.07 -0.69,-1.58 -0.12,-0.51 -0.2,-1.09 -0.12,-1.7 0.08,-0.61 0.33,-1.24 0.69,-1.91 0.37,-0.66 0.85,-1.36 1.21,-2 0.36,-0.64 0.58,-1.22 0.71,-1.89 0.13,-0.67 0.16,-1.44 0,-2.1 -0.16,-0.66 -0.52,-1.23 -0.92,-1.9 -0.41,-0.66 -0.86,-1.42 -1.17,-2.13z"
            fill="#FAFAFA"
          />
        </svg>
      </div>
    </div>
    <svg
      aria-hidden="true"
      class="c14"
      fill="none"
      height="100"
      viewBox="0 0 100 100"
      width="100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 50C0 22.3858 22.3858 0 50 0C77.6142 0 100 22.3858 100 50C100 77.6142 77.6142 100 50 100H0L0 50Z"
        fill="#334BC8"
      />
    </svg>
  </div>
</header>
`;
